In [29]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [30]:
import numpy as np
import pandas as pd

나눠진 csv파일을 한번에 불러왔습니다.

In [31]:
df1 = '/content/drive/MyDrive/nlp pj/talksets_train_1.csv'
df2 = '/content/drive/MyDrive/nlp pj/talksets_train_2.csv'
df3 = '/content/drive/MyDrive/nlp pj/talksets_train_3.csv'
df4 = '/content/drive/MyDrive/nlp pj/talksets_train_4.csv'
df5 = '/content/drive/MyDrive/nlp pj/talksets_train_5.csv'

In [32]:
df = pd.concat(
   map(pd.read_csv, [df1, df2, df3, df4, df5]), ignore_index=True)
print(df)

                               text             types  is_immoral  intensity
0       부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네       ['CENSURE']        True        2.0
1                 맞아. 사람들이 진짜 개념이없다       ['CENSURE']        True        2.0
2                      저렇게는 살지 말아야지       ['CENSURE']        True        2.0
3            그러게 게으르고 멍청한 사람들은 맞아야해       ['CENSURE']        True        2.0
4                 인방 보는 남자는 거르는게 맞다       ['CENSURE']        True        1.0
...                             ...               ...         ...        ...
363149      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는       ['CENSURE']        True        1.4
363150                   오죽 하면 그러겠나  ['IMMORAL_NONE']       False        0.0
363151              너 얼굴 보면 한숨 나온다.       ['CENSURE']        True        1.6
363152                      신경 끄세요.  ['IMMORAL_NONE']       False        0.0
363153                이제라도 관리 좀 해라.  ['IMMORAL_NONE']       False        0.0

[363154 rows x 4 columns]


스마일 게이트 데이터와 합치기 위해 컬럼명을 통일했습니다.

In [33]:
df['문장'] = df['text']

필요없는 컬럼을 삭제하고, 새로 컬럼을 생성합니다.

In [34]:
df.drop('text',axis=1, inplace=True)
df.drop('types',axis=1, inplace=True)
df.drop('intensity',axis=1, inplace=True)

In [35]:
df['clean'] = 0

In [36]:
df['violence'] = 0

In [37]:
df

is_immoral                           문장  clean  violence
0             True  부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네      0         0
1             True            맞아. 사람들이 진짜 개념이없다      0         0
2             True                 저렇게는 살지 말아야지      0         0
3             True       그러게 게으르고 멍청한 사람들은 맞아야해      0         0
4             True            인방 보는 남자는 거르는게 맞다      0         0
...            ...                          ...    ...       ...
363149        True      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는      0         0
363150       False                   오죽 하면 그러겠나      0         0
363151        True              너 얼굴 보면 한숨 나온다.      0         0
363152       False                      신경 끄세요.      0         0
363153       False                이제라도 관리 좀 해라.      0         0

[363154 rows x 4 columns]

True와 False 를 숫자 데이터로 변경해줍니다.

스마일게이트의 데이터 셋에 맞추어 float타입으로 변경합니다.

In [38]:
df['is_immoral'] = df['is_immoral'].astype(float)

In [39]:
df['violence'] = df['is_immoral']

In [40]:
df

is_immoral                           문장  clean  violence
0              1.0  부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네      0       1.0
1              1.0            맞아. 사람들이 진짜 개념이없다      0       1.0
2              1.0                 저렇게는 살지 말아야지      0       1.0
3              1.0       그러게 게으르고 멍청한 사람들은 맞아야해      0       1.0
4              1.0            인방 보는 남자는 거르는게 맞다      0       1.0
...            ...                          ...    ...       ...
363149         1.0      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는      0       1.0
363150         0.0                   오죽 하면 그러겠나      0       0.0
363151         1.0              너 얼굴 보면 한숨 나온다.      0       1.0
363152         0.0                      신경 끄세요.      0       0.0
363153         0.0                이제라도 관리 좀 해라.      0       0.0

[363154 rows x 4 columns]

violence값이 0인 데이터는 clean한 데이터이기에 clean 값에 1을 넣어줍니다.

In [41]:
for i in df[(df['violence'] == 0)].index:
        df.at[i, 'clean'] = 1

In [42]:
df

is_immoral                           문장  clean  violence
0              1.0  부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네      0       1.0
1              1.0            맞아. 사람들이 진짜 개념이없다      0       1.0
2              1.0                 저렇게는 살지 말아야지      0       1.0
3              1.0       그러게 게으르고 멍청한 사람들은 맞아야해      0       1.0
4              1.0            인방 보는 남자는 거르는게 맞다      0       1.0
...            ...                          ...    ...       ...
363149         1.0      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는      0       1.0
363150         0.0                   오죽 하면 그러겠나      1       0.0
363151         1.0              너 얼굴 보면 한숨 나온다.      0       1.0
363152         0.0                      신경 끄세요.      1       0.0
363153         0.0                이제라도 관리 좀 해라.      1       0.0

[363154 rows x 4 columns]

In [43]:
df['clean'] = df['clean'].astype(float)

In [44]:
df

is_immoral                           문장  clean  violence
0              1.0  부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네    0.0       1.0
1              1.0            맞아. 사람들이 진짜 개념이없다    0.0       1.0
2              1.0                 저렇게는 살지 말아야지    0.0       1.0
3              1.0       그러게 게으르고 멍청한 사람들은 맞아야해    0.0       1.0
4              1.0            인방 보는 남자는 거르는게 맞다    0.0       1.0
...            ...                          ...    ...       ...
363149         1.0      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는    0.0       1.0
363150         0.0                   오죽 하면 그러겠나    1.0       0.0
363151         1.0              너 얼굴 보면 한숨 나온다.    0.0       1.0
363152         0.0                      신경 끄세요.    1.0       0.0
363153         0.0                이제라도 관리 좀 해라.    1.0       0.0

[363154 rows x 4 columns]

총 363154개의 데이터 중 약 40%의 데이터가 clean데이터로 구성되어 있어 비율을 유지하기 위해 2만개의 40%인 8000개의 데이터를 clean 데이터에서 뽑아줍니다.

In [45]:
df.drop('is_immoral',axis=1, inplace=True)

In [46]:
df

문장  clean  violence
0       부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네    0.0       1.0
1                 맞아. 사람들이 진짜 개념이없다    0.0       1.0
2                      저렇게는 살지 말아야지    0.0       1.0
3            그러게 게으르고 멍청한 사람들은 맞아야해    0.0       1.0
4                 인방 보는 남자는 거르는게 맞다    0.0       1.0
...                             ...    ...       ...
363149      뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는    0.0       1.0
363150                   오죽 하면 그러겠나    1.0       0.0
363151              너 얼굴 보면 한숨 나온다.    0.0       1.0
363152                      신경 끄세요.    1.0       0.0
363153                이제라도 관리 좀 해라.    1.0       0.0

[363154 rows x 3 columns]

In [47]:
condition = (df.clean == 1)
df_clean = df[condition]

In [48]:
df_clean

문장  clean  violence
10                    쉴드가 아니라 국가가 면제해준거야.    1.0       0.0
17                          짐승입장도 들어봐야지ㅋㅋ    1.0       0.0
19      #@인간및인간집단.인물.정치인.남성#이 이제 급하긴 한가보네    1.0       0.0
20                         깐부라고 부르기까지 하다니    1.0       0.0
22            어머니 간 이식 해 주기 싫어서 도망쳤다고...?    1.0       0.0
...                                   ...    ...       ...
363144            와 넌 모욕죄 낼 통장 잔고가 넉넉한가봐?    1.0       0.0
363145                   바나나 먹으면 나한테 반하나?    1.0       0.0
363150                         오죽 하면 그러겠나    1.0       0.0
363152                            신경 끄세요.    1.0       0.0
363153                      이제라도 관리 좀 해라.    1.0       0.0

[162948 rows x 3 columns]

In [49]:
df_cfin = df_clean.sample(n=8000, random_state=7)

In [50]:
df_cfin23 = df_clean.sample(n=81474, random_state=7)

In [51]:
df_cfin

문장  clean  violence
263408                   웅얼거리지 말고 크게 말해!    1.0       0.0
260632                  핸드폰으로 좀 할 게 있어서.    1.0       0.0
165408                         무시를 왜 당해?    1.0       0.0
294186                  나 이번 추석때 사이판 간다~    1.0       0.0
330155  엌ㅋㅋ나만 그런 줄 알고 자괴감 들었는데. 당연한 거였네?    1.0       0.0
...                                  ...    ...       ...
174090              쟤 아까부터 왜이렇게 풀이 죽어있어?    1.0       0.0
82922                너무 높아서 올라가는데도 한참이겠다    1.0       0.0
186971  와 너네 아빠 무조건 명절엔 시골 가자고 하셨던 분 아냐?    1.0       0.0
146283                  일을 못 하면 시키는 것만 해    1.0       0.0
301591                      아저씨 침 뱉지 마세요    1.0       0.0

[8000 rows x 3 columns]

In [52]:
condition = (df.violence == 1)
df_violence = df[condition]

In [53]:
df_violence

문장  clean  violence
0           부랴부랴 왔는데 아무도 안왔네. 시간개념들이 없네    0.0       1.0
1                     맞아. 사람들이 진짜 개념이없다    0.0       1.0
2                          저렇게는 살지 말아야지    0.0       1.0
3                그러게 게으르고 멍청한 사람들은 맞아야해    0.0       1.0
4                     인방 보는 남자는 거르는게 맞다    0.0       1.0
...                                 ...    ...       ...
363146                        역겨우니까 하지마    0.0       1.0
363147  나 이제부터 애교 절대 안 해. 해달라고 하기만 해. 확    0.0       1.0
363148                 죽으려면 그냥 죽지 왜 고민함    0.0       1.0
363149          뷔응신처럼 죽지도 못하면서 괜히 뛰어들기는    0.0       1.0
363151                  너 얼굴 보면 한숨 나온다.    0.0       1.0

[200206 rows x 3 columns]

In [54]:
df_vfin23 = df_violence.sample(n=100103, random_state=7)

In [55]:
df_vfin = df_violence.sample(n=12000, random_state=7)

In [56]:
df_vfin

문장  clean  violence
186727                     엄마가 헤어지래서 헤어졌다는 게 사실이냐 찌질아? ㅋㅋ    0.0       1.0
202950  돈도 없고 지능 딸리고 ㅋㅋ 안 봐도 그런샛기들이 얼마나 밑바닥 인생일지 바로 알 ...    0.0       1.0
106936            그동안 쓰레기 같은 기사 써서 멀쩡한 사람 망쳐 놓은 건 다 잊었나 봄    0.0       1.0
168332       그래도 결이 좀 다르지. 나는 아마 혼혈일 걸? 순수 똥양이랑 동급 취급하지마.    0.0       1.0
149244                                 여자들은 반려묘 못키우게 해야 해    0.0       1.0
...                                                   ...    ...       ...
192491  결국 남은 건 막대한 손해에 아줌마 #@인간및인간집단.인물.인물_무정보.여성#, 한...    0.0       1.0
227442                         아름다운 추억 본인이 깽판쳐 놓고 감성팔이 마라    0.0       1.0
6022                                #@#이 바퀴벌레보다 더 대단한 건가?    0.0       1.0
58102           니 말하는꼬라지 보니까 딱 못배운 공익티 팍팍내내  정신공익이냐?ㅋㅋㅋㅋㅋ    0.0       1.0
69414                               돈도 많은 것들이 추접하게 무슨 짓이람    0.0       1.0

[12000 rows x 3 columns]

In [57]:
df.to_csv("immoral.csv", index=False, encoding="utf-8-sig")

In [58]:
df_cfin23.to_csv("immoral_23c.csv", index=False, encoding="utf-8-sig")
df_vfin23.to_csv("immoral_23.csv", index=False, encoding="utf-8-sig" )

In [59]:
df_cfin.to_csv("immoral_cfin.csv", index=False, encoding="utf-8-sig")
df_vfin.to_csv("immoral_vfin.csv", index=False, encoding="utf-8-sig")